hedera-node/hedera-mono-service/src/main/java/com/hedera/services/txns/contract/ContractCreateTransitionLogic.java

```diff
@@ -214,6 +232,11 @@ public void doStateTransitionOperation(
 
         if (!result.isSuccessful()) {
             worldState.reclaimContractId();
+
+            // Unlink unneeded alias from the alias map to free memory
```

Comment:
Leaving this alias linked would have worse side effects than just its memory footprint; if the transaction doesn't succeed, the alias isn't persisted to the serialized state---i.e., the `accounts` `MerkleMap`.

This means if a node reconnects (and thus reconstructs its `aliases` map from the serialized state), it will have different aliases than nodes that didn't reconnect. This could cause an ISS.

⚠️ There is also a more subtle timing issue involving signature verification, that means after L278 we need to mark the alias changed in the `SigImpactHistorian`,
```
    sigImpactHistorian.markEntityChanged(newContractId.getContractNum());
    if (relayerId != null) {
        sigImpactHistorian.markAliasChanged(
            ByteStringUtils.wrapUnsafely(newContractAddress.toArrayUnsafe()));
    }
  
```